SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/OptOutServices/EmailServiceV1.swift

```diff
@@ -0,0 +1,209 @@
+    public let name: String
+    public let value: String
+}
+
+public struct DeleteEmailDataRequestV1: Codable {
+    public let items: [EmailDataRequestItemV1]
+
+    public init(items: [EmailDataRequestItemV1]) {
+        self.items = items
+    }
+}
+
+public protocol EmailServiceV1Protocol {
+    func fetchEmailData(items: [EmailDataRequestItemV1]) async throws -> EmailDataResponseV1
+    func deleteEmailData(items: [EmailDataRequestItemV1]) async throws
+}
+
+public struct EmailServiceV1: EmailServiceV1Protocol {
+    public struct Constants {
+        static let endpointSubPath = "/dbp/em/v1"
+        static let maxBatchSize = 100
+    }
+
+    public let urlSession: URLSession
+    private let authenticationManager: DataBrokerProtectionAuthenticationManaging
+    private let settings: DataBrokerProtectionSettings
+    private let servicePixel: DataBrokerProtectionBackendServicePixels
+
+    public init(urlSession: URLSession = URLSession.shared,
+                authenticationManager: DataBrokerProtectionAuthenticationManaging,
+                settings: DataBrokerProtectionSettings,
+                servicePixel: DataBrokerProtectionBackendServicePixels) {
+        self.urlSession = urlSession
+        self.authenticationManager = authenticationManager
+        self.settings = settings
+        self.servicePixel = servicePixel
+    }
+
+    public func fetchEmailData(items: [EmailDataRequestItemV1]) async throws -> EmailDataResponseV1 {
+        guard !items.isEmpty else {
+            throw EmailErrorV1.noEmailData
+        }
+
+        guard items.count <= Constants.maxBatchSize else {
+            throw EmailErrorV1.batchSizeExceeded
+        }
+
+        var urlComponents = URLComponents(url: settings.endpointURL, resolvingAgainstBaseURL: true)
+        urlComponents?.path += "\(Constants.endpointSubPath)/email-data"
+
+        guard let url = urlComponents?.url else {
+            throw EmailErrorV1.cantGenerateURL
+        }
+
+        var request = URLRequest(url: url)
+        guard let authHeader = await authenticationManager.getAuthHeader() else {
+            servicePixel.fireEmptyAccessToken(callSite: .fetchEmailData)
+            throw AuthenticationError.noAuthToken
+        }
+
+        request.setValue(authHeader, forHTTPHeaderField: "Authorization")
+        request.addValue("application/json; charset=utf-8", forHTTPHeaderField: "Content-Type")
+        request.httpMethod = "POST"
+
+        let requestBody = EmailDataRequestV1(items: items)
+        request.httpBody = try JSONEncoder().encode(requestBody)
+
+        let (data, response) = try await urlSession.data(for: request)
+        try validateHTTPResponse(response)
+
+        do {
+            return try JSONDecoder().decode(EmailDataResponseV1.self, from: data)
+        } catch {
+            throw EmailErrorV1.invalidResponse
+        }
+    }
+
+    public func deleteEmailData(items: [EmailDataRequestItemV1]) async throws {
```

Comment:
Does the deletion end point have a number of items limit too?

Reply:
No items limit for this end point